packages/opentelemetry-exporter-collector/src/platform/browser/CollectorExporterBrowserBase.ts

```diff
@@ -61,20 +61,41 @@ export abstract class CollectorExporterBrowserBase<
     onSuccess: () => void,
     onError: (error: collectorTypes.CollectorExporterError) => void
   ) {
+    if (this._isShutdown) {
+      this.logger.debug('Shutdown already started. Cannot send objects');
+      return;
+    }
     const serviceRequest = this.convert(items);
     const body = JSON.stringify(serviceRequest);
 
-    if (this._useXHR) {
-      sendWithXhr(
-        body,
-        this.url,
-        this._headers,
-        this.logger,
-        onSuccess,
-        onError
-      );
-    } else {
-      sendWithBeacon(body, this.url, this.logger, onSuccess, onError);
-    }
+    const promise = new Promise(resolve => {
```

Comment:
Can we change this to be async/await?

Reply:
We can't this function is a promise on purpose. The collector might call send many times, every times it is called the promise is created and added to the _sendingPromises (removed after the the promise is resolved). When you call shutdown it will stop all new send but then will wait for all previous send and when they finish the shutdown promise will be resolved. This way all the things in progress that are being sent will be able to finish and then notify correctly the shutdown.

Reply:
Same as above I think this would have been achievable with async/await, but if you prefer this style that is fine.